cmd/gaia/init/init.go

```diff
@@ -51,25 +51,32 @@ func InitCmd(ctx *server.Context, cdc *codec.Codec, appInit server.AppInit) *cob
 		RunE: func(_ *cobra.Command, _ []string) error {
 			config := ctx.Config
 			config.SetRoot(viper.GetString(cli.HomeFlag))
+
 			chainID := viper.GetString(client.FlagChainID)
 			if chainID == "" {
 				chainID = fmt.Sprintf("test-chain-%v", common.RandStr(6))
 			}
+
 			nodeID, _, err := InitializeNodeValidatorFiles(config)
 			if err != nil {
 				return err
 			}
 
-			if viper.GetString(flagMoniker) != "" {
-				config.Moniker = viper.GetString(flagMoniker)
+			moniker := viper.GetString(flagMoniker)
+			if moniker == "" {
```

Comment:
No need to validate manually. Just add

```golang
cmd.MarkFlagRequired(flagMoniker)
```

Reply:
When did they add this??

Reply:
Unsure, certainly we have used it somewhere else in our code.